devito/types/dense.py

```diff
@@ -1311,3 +1312,198 @@ def parent(self):
         return self._parent
 
     _pickle_kwargs = Function._pickle_kwargs + ['parent']
+
+
+class SeparableFunction(Differentiable):
+    """
+    A function separable in its dimensions, ie f(x,y,z) = f(x)*f(y)*f(z)
```

Comment:
I'm not gonna repeat the usual things over and over again but... **keywords should always be capitalized**

function -> Function
dimension -> Dimension

and yes, we also do plurals:

functions -> Functions

etc

Reply:
ie -> i.e.